### DISCUSSION TITLE
Bug: query params doesn't work?

### DISCUSSION THREAD

**Discussion by thompson0012** [litestar-org/litestar#2816] [Q&A] [answered]
### Description

i am trying the litestar library for the api development. i found there are some stranges

doc: https://docs.litestar.dev/latest/usage/routing/parameters.html#query-parameters

```python
from litestar import Litestar, get


@get("/", sync_to_thread=False)
def index(param: str) -> dict[str, str]:
    return {"param": param}


app = Litestar(route_handlers=[index])
```

```shell
> curl http://127.0.0.1:8000/?param=foo
{"param":"foo"}
> curl http://127.0.0.1:8000/?param=bar
{"param":"bar"}
```

but the terminal show me when i follow the instruction
```shell
zsh: no matches found: http://localhost:8000?param=uiyhkjh
```

when i remove the query param, it showed
```shell
{"status_code":400,"detail":"Missing required query parameter 'param' for url http://localhost:8000/"}%      
```

i have no idea on the issue, i just simply follow the guidelines, without any modification

### MCVE

```python
# Your MCVE code here
```

### Steps to reproduce

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
```

### Screenshots

```bash
"![SCREENSHOT_DESCRIPTION](SCREENSHOT_LINK.png)"
```

### Litestar Version

Version(major=2, minor=4, patch=1, release_level='final', serial=0)

### Platform

- [ ] Linux
- [X] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh Litestar dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/2815">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2815/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/2815/pledge.svg">
</picture>
</a>

**Comment by provinzkraut** [comment]
> but the terminal show me when i follow the instruction

It's not clear what you are doing here. Where does it show that? What instruction are you following?

The `curl` command you're running seems to work fine from the output?

**Reply by thompson0012** [reply]
i am trying to repeat the steps from the guideline. 

i simply copy and paste the command 

```python
from litestar import Litestar, get


@get("/", sync_to_thread=False)
def index(param: str) -> dict[str, str]:
    return {"param": param}


app = Litestar(route_handlers=[index])
```
and execute it in shell

```shell
> curl http://localhost:8000?param=uiyhkjh
```

but the curl resulted in the following

```shell
zsh: no matches found: http://localhost:8000?param=uiyhkjh
```

**Reply by thompson0012** [reply]
<img width="947" alt="Screenshot 2023-12-01 at 23 37 25" src="https://github.com/litestar-org/litestar/assets/51963680/0acf5488-4199-41ff-bbeb-bd6bfbb45b30">

**Reply by provinzkraut** [reply, marked as answer]
The error message is coming from zsh, not Litestar. You need to either escape the `?` or surround the whole string in quotes.

`curl "http://localhost:8000?param=uiyhkjh"`

**Reply by thompson0012** [reply]
ok, got it. thanks!
